.github/workflows/libxrpl.yml

```diff
@@ -0,0 +1,86 @@
+name: Upload libXRPL to Conan
+env:
+  CONAN_URL: http://18.143.149.228:8081/artifactory/api/conan/conan-non-prod
+  CONAN_LOGIN_USERNAME_RIPPLE: ${{ secrets.CONAN_USERNAME }}
+  CONAN_PASSWORD_RIPPLE: ${{ secrets.CONAN_TOKEN }}
+on:
+  pull_request:
+    paths:
+      - 'src/ripple/protocol/impl/BuildInfo.cpp'
+      - '.github/workflows/libxrpl.yml'
+    branches: [release]
```

Comment:
- [Here](https://github.com/XRPLF/rippled/pull/4968) is a patch release PR that targeted `master`.
- [Here](https://github.com/XRPLF/rippled/pull/4870) is a major release PR that targeted `develop`.

Do we channel all of our releases through a single branch? I don't know. Even if we do, I'm not sure we can count on it always happening through a PR. I understand you want to publish a package before the commit is pushed to a protected branch or tagged. Can we get an expert on the release process to chime in? @seelabs?

Once we determine the right condition to check, we can think about how to deduplicate some of the steps here. Maybe we can move this workflow into a single conditional step at the end of the test workflow.

Reply:
Would be indeed useful to know if `release` is supposed to always be used for this purpose @seelabs 👍 

When implementing this i deliberately sacrificed duplication for non-intrusiveness. I agree we can deduplicate quite a bit but i thought we would be better off creating a separate issue/PR for that. Let me know what you think @thejohnfreeman @seelabs .

Reply:
I've added a workflow in a [branch](https://github.com/thejohnfreeman/rippled/actions/runs/9484998818/job/26139475923) that builds the Conan package locally and tests that an example downstream package can import it successfully. We could add a job to upload the recipe (and binary?) conditional on that test (and whatever conditions we decide determine a release).

Reply:
Not sure how useful it is to build an "example" package. Unless you have a package that uses everything libxrpl provides of course, in which case that's great. 
I only really care that Clio (the only real/important consumer afaik) still works so I'd certainly prefer to leave the notification step that triggers Clio's CI. If you want to add an example package build in between - i have no problem with that. 
I personally prefer to get simple things like CI out of the way and functional quicker and then improve if needed later.

Reply:
It is useful to test that, at least, `libxrpl` can be found and linked by a dependent package. It is possible to build a Conan package that doesn't actually export anything that can be linked.

Reply:
Clio could be our example package, instead, if we want to make it so. But the test that Clio can build against the `xrpl` Conan package must be moved to this project, so that it is a blocker for our commits. The example package I wrote is much simpler, but it is in this project. That is the key attribute.

Reply:
Well, i personally don't like the idea of building Clio from this repo - that's as intrusive as it gets. But i do see your points. Please do what you want with this PR or let's discuss next steps next week.

Reply:
@seelabs can we get your input on the release process please?

Reply:
@thejohnfreeman @godexsoft: 
 
You can rely on releases and release candidates always being pushed to `release`. In most cases, you can rely on betas being pushed to `release` as well, but that may not happen if there are `develop` has diverged from `release` and we aren't merging betas into release until we transition the release candidate to release.

Here's the process that happens when `develop` and `master` have not diverged (usually the branches have not diverged, but this can happen for hot fixes or when a rc is active and we merge commits to develop that shouldn't be in the rc):

* betas and release candidates are merged to `develop` and `release`
* releases are merged to `develop`, `release` and `master`
 
I understand that happened with the old process when `develop` and `master` diverged, but it's less clear to me what happens in the new process. Here's my understanding for the new process if release and develop have diverged:

* betas to to `develop`, but not master
* release candidates to to `release` but not `develop`
* once a release candidate transitions to a release, we merge the `release` branch into `develop` with a merge commit.
* once the reverse happens, new betas will be merged to `master` again until the branches diverge

We usually open beta PRs and rc PRs against the `release` branch, but sometimes the PR is accidentally opened against the `develop` branch and if the branches haven't diverged there's no harm.

Reply:
It sounds like the two important conditions are:

1. All releases will pass through the `release` branch.
2. For Clio to be notified before a release actually lands, the push to `release` must first pass through a PR.

In that case, the conditions of the workflow in this PR are already correct.

I think the workflows are ripe for a re-factoring after this, but it doesn't need to be a blocker.

Reply:
If everyone are happy we can merge this but we will need to setup an account (and a token from that) with access to execute the remote repo workflow.

Reply:
Are you sure a new account is required? I think when you create the PAT, you can just set the "resource owner" field to the XRPLF organization and limit its scope to the clio respository.

Reply:
No, it's not required. But if i create a PAT from say my account it will probably use my account when creating PRs/issues from that workflow.
It's the easiest option.. but perhaps having some "XRPLD[bot]" account would be preferred?

Reply:
or would setting "resource owner" change the PR/issue author, do you know?

Reply:
I don't know. It can be tested though. Is there someone with organization permissions who we can delegate this to?

Reply:
I have evidence that it will still use the `github-actions` account to create the PR even if i use a PAT created for my account. I can create one and ask @ximinez to set it up in this repo.

Reply:
Not all releases are guaranteed to pass through `release`. Specifically, a hotfix (e.g. [2.2.1](https://github.com/XRPLF/rippled/pull/5078)) can go directly to `master` without touching `develop` or `release` until it is backported. Also the `paths` and `branches` conditions [are essentially `AND`ed together](https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#running-your-pull_request-workflow-based-on-the-head-or-base-branch-of-a-pull-request). No changes should ever get to `release` (or `master` for that matter) without the version number changing in `BuildInfo.cpp`, which would make that check redundant. Conversely, `BuildInfo.cpp` is almost never changed except to change the version number.

This contradicts what @seelabs said above. Take the example of 2.2.1, the reason is simple: The issue wasn't discovered in 2.2.0 until well after the first 2.3.0 beta had been merged to `release`. At that point, the only options were either roll back `release` (very bad), or let `master` and `release` diverge (not so bad).

Therefore, I suggest either:
* Remove `paths`, and change `branches` to use both `release` and `master`. 
* Remove `branches`, and leave `paths` as-is.

Both risk adding extra builds, but drastically reduce the chances of missing a build, which seems like it would be worse.